nixui/nix_eval.py

```diff
@@ -24,22 +24,41 @@ def nix_instantiate_eval(expr, strict=False):
 
 def get_modules_defined_attrs(module_path, attr_loc=[]):
     attributes = {}
-    for sub_attr_datapoint in get_subattribute_data(module_path, attr_loc):
-        full_attr_loc = attr_loc + [sub_attr_datapoint['name']]
-        if sub_attr_datapoint['position'] is not None and sub_attr_datapoint['position']['file'] == module_path:
-            if sub_attr_datapoint['type'] != 'set':
-                attributes[tuple(full_attr_loc)] = sub_attr_datapoint
-            else:
-                res = get_modules_defined_attrs(
-                    module_path,
-                    full_attr_loc
-                )
-                if not res:
-                    attributes[tuple(full_attr_loc)] = sub_attr_datapoint
-                else:
-                    attributes.update(res)
-    return attributes
 
+    return {
+        tuple(v['name']): {"position": v['position']}
+        for v in nix_instantiate_eval("""
+let
+  config = import <nixos-config> {config = {}; pkgs = import <nixpkgs> {}; lib = import <nixpkgs/lib>;};
```

Comment:
I'm not sure we should hardcode `nixos-config` here. Eventually we'll want to 
- 1) get attribute positions for each module within `imports` (excluding those within `nixpkgs`, e.g. `"${nixpkgs}/nixos/modules/installer/cd-dvd/iso-image.nix"`)
- 2) get attribute positions for non-configuration files where users want to modify the configurations of other build types much further down the line (e.g. https://github.com/lapp0/nix-gui/issues/23)

Reply:
See 9bd09a41c8f3927c6e8c98e30a7ec7cf07c789a3